src/types.ts

```diff
@@ -249,6 +253,25 @@ export interface Instance {
   };
 }
 
+export interface UploadJobStatus {
+  [jobId: string]: {
+    isCompleted: boolean,
+    result: {
+      url: string,
+      hash: string
+    }
+    failed: boolean
```

Comment:
do we need both `failed` and `isCompleted` ? Is it possible to complete, but fail?, or fail and still complete?

Reply:
it's possible that a job is still running. `failed` would be false, but so would `isCompleted`.